lib/sync/installation.js

```diff
@@ -81,7 +81,10 @@ module.exports.processInstallation = (app, queues) => {
     const github = await getEnhancedGitHub(installationId)
 
     const nextTask = getNextTask(subscription)
-    if (!nextTask) return
+    if (!nextTask) {
+      await subscription.update({ syncStatus: 'COMPLETE' })
```

Comment:
I don't understand where this came from. It didn't seem to be in the worker/index.js... so why do we need it now?

Reply:
This is a new behaviour intended to resolve the following pain point:

1. Customer installs Jira integration
2. Initial sync completes all work but fails while updating sync to COMPLETE
3. We fix bug and retrigger sync to correct problem

Prior to this change, the sync would remain FAILED even after a sync was retriggered. This ensures that if a sync runs and there is no remaining tasks to complete, we mark it as COMPLETE.

Reply:
Sounds like a fairly uncommon edge case. I would recommend adding a log statement here so that we can see how often we're running into it. Or if/when we're doing any sleuthing for a particular user. It's not much, but a little visibility.

Reply:
We could also ping datadog so we can see this over time. Nice edge case!

Reply:
This case happens when the customer triggers a "normal" sync or, more likely, _we_ trigger the sync on their behalf.